src/app/shared/provider.service.ts

```diff
@@ -17,11 +17,11 @@
 export class ProviderService {
 
   /* set up project provider */
-  setUpProvider(tool) {
+  setUpProvider(tool, version) {
```

Comment:
from above, it seems like this method is often called as `setupProvider(tool, null)`
It would have been simpler (and more readable) to provide an overridden `setupProvider(tool) { setupProvider(tool, null)}` if thats possible in typescript. You may have IDE support for this refactoring

Reply:
Would default parameters work too? So default version to null.